reVX/setbacks/base.py

```diff
@@ -0,0 +1,658 @@
+# -*- coding: utf-8 -*-
+"""
+Compute setbacks exclusions
+"""
+from abc import ABC
+from concurrent.futures import as_completed
+from warnings import warn
+import os
+import logging
+import pathlib
+import numpy as np
+import geopandas as gpd
+from rasterio import features
+from shapely.geometry import shape
+
+from rex.utilities import parse_table, SpawnProcessPool, log_mem
+from reV.handlers.exclusions import ExclusionLayers
+from reVX.utilities.exclusions_converter import ExclusionsConverter
+from reVX.utilities.utilities import log_versions
+
+logger = logging.getLogger(__name__)
+
+
+class BaseSetbacks(ABC):
+    """
+    Create exclusions layers for setbacks
+    """
+
+    def __init__(self, excl_fpath, plant_height, regulations_fpath=None,
+                 multiplier=None, hsds=False, chunks=(128, 128)):
+        """
+        Parameters
+        ----------
+        excl_fpath : str
+            Path to .h5 file containing exclusion layers, will also be the
+            location of any new setback layers
+        plant_height : float | int
+            Plant height (m), used to determine setback distance.
+        regulations_fpath : str | None, optional
+            Path to regulations .csv file, if None create generic
+            setbacks using max-tip height * "multiplier", by default None
```

Comment:
Can we describe what this regulations_fpath csv should look like? What are the required columns and what does each row represent? County, State, Value Type, Value? Does it need the FIPS fields or is that optional? Even an incomplete description would be useful.

Reply:
Haha yeah, definitely not your fault but reading through i've been asking myself what the hell are these variables?? We can't totally prevent the inheritance of technical debt but we can try and improve it. Thanks!